pkg/api/validation/validation.go

```diff
@@ -468,6 +473,28 @@ func validateGlusterfs(glusterfs *api.GlusterfsVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+var validMetadataFieldPathExpressions = util.NewStringSet("metadata.name", "metadata.namespace", "metadata.labels", "metadata.annotations")
```

Comment:
Actually, the volume plugin won't support `status.podIP`.  I think you should remove it from the other descriptions of the downward API since there is a race condition with that currently, and we'll add it to to the volume plugin later.